### ISSUE TITLE
Regexp $-commands and ^-listen with ':'

### ISSUE THREAD

**Issue by brazilofmux** [brazilofmux/tinymux#662]
Original [issue 660](https://code.google.com/p/tinymux/issues/detail?id=660) created by brazilofmux on 2011-05-04T20:58:10.000Z:

TinyMUX 2.10

&gt; &amp;test.cmd me=$+test:@pemit me=Foo
&gt; @set me/test.cmd=regexp
&gt; @set me=COMMANDS

&gt; +test
Foo
&gt; &amp;test.cmd me=$+test::@pemit me=Foo
Brazil @pemit me=Foo
&gt; &amp;test.cmd me=$+test::@pemit me=Foo
Huh?
&gt; &amp;test.cmd me=$+test\x3A:@pemit me=Foo
&gt; +test:
Foo
&gt; &amp;test.cmd me=$+test(?&gt;\x3A)?:@pemit me=Foo
&gt; +test:
Foo
&gt; +test
Foo

Basically, ':' cannot appear in the command except as \x3A because TinyMUX parses it as the end of the command pattern.

'\x3A' can be used instead of ':'. '(?&gt;...)' can be used instead '(?:...)' in most cases, but it isn't the same.

On PennMUSH, ':' is used to escape the training ':'.

**Comment by brazilofmux** [comment]
Fixed in commit 2edfad9c on the brazil branch.

Adopted the PennMUSH convention: `::` in a `$`-command or `^`-listen pattern escapes a literal `:`. The first single `:` (not part of a `::` pair) remains the pattern/action delimiter. After splitting, `::` is collapsed to `:` in the pattern before matching.

**Examples:**
- `$+test:::@pemit me=Foo` — pattern `+test:`, action `@pemit me=Foo`
- `$(?::red|blue) ball:@pemit %#=Match` — regexp non-capturing group works

**Changes:**
- Added `find_pattern_delimiter()` and `unescape_pattern_colons()` in `stringutil.cpp`
- Updated all 4 call sites: `atr_match1()`, `Hearer()`, `Commer()`, `sweep_check()`
- Documented `::` escaping in `help arbitrary commands`

Fully backward-compatible — existing patterns without `::` work identically. All 290 smoke tests pass.
